telemetry/src/main/java/com/newrelic/telemetry/Backoff.java

```diff
@@ -0,0 +1,38 @@
+package com.newrelic.telemetry;
+
+public class Backoff {
+
+  private final long maxBackoffTimeMs;
+  private final long backoffFactorMs;
+  private final int maxRetries;
+  private int numRetries = 0;
+
+  public Backoff(long maxBackoffTimeMs, long backoffFactorMs, int maxRetries) {
+    this.maxBackoffTimeMs = maxBackoffTimeMs;
+    this.backoffFactorMs = backoffFactorMs;
+    this.maxRetries = maxRetries;
+  }
+
+  public static Backoff defaultBackoff() {
+    return new Backoff(15000, 1000, 10);
```

Comment:
whoa. this is hard to read. I wonder if we want durations or something to make this clearer?

Reply:
or maybe some named parameters.  ;)

Reply:
I made a builder that is as close as we can get to named params in this stuff.  :)